content/blog/deploy-stateful-mongodb-application-on-kubernetes-clusters-in-hpe-greenlake-for-containers-1.md

```diff
@@ -0,0 +1,543 @@
+
+This blog post describes the process of deploying the stateful MongoDB application on the created Kubernetes clusters in HPE GreenLake for Containers. Using the Kubernetes StatefulSet and Headless Service, together with the pre-provisioned persistent volumes, the MongoDB application can be deployed as a Replica Set that provides redundancy, fault tolerance and high availability. This MongoDB application deployment can be used for working with a large amount of data and high loads in customer production environments.
+
+## MongoDB Application
+
+[MongoDB](https://www.mongodb.com/) is a general-purpose, document-based NoSQL database program which is a popular alternative to traditional databases. The MongoDB data model can represent any kind of object structures which can have properties or even be nested for multiple levels. Unlike the relational databases, MongoDB can store large amounts of data without requiring a logical category or schema. Therefore, it takes up the challenge of focusing on scalability and the speed of queries. MongoDB is free, open-source software. You can download the MongoDB packages, set them up and configure them at no expense.
+
+MongoDB can be deployed as a standalone instance that is running as a single *mongod* server. It can also be deployed as a [Replica Set](https://www.mongodb.com/docs/manual/core/replica-set-architectures) with multiple running *mongod* instances that maintain the same data set. The Replica Set contains several data-bearing nodes. Of those data-bearing nodes, one and only one member is deemed the *Primary* node, while the other nodes are deemed *Secondary* nodes.
+
+![](/img/mongodb-replica-set.png)
+
+The standalone MongoDB instance deployment is suitable for testing and some aspects of development. However, it’s not adequate for production use. The MongoDB Replica Set deployment ensures multiple replicas running at any given time, which provides redundancy, fault tolerance and high availability. It is recommended for production environments, such as HPE GreenLake for Containers. The blog post will focus on deploying the MongoDB application as a Replica Set.
+
+## Deploy MongoDB Application
+
+### Requirements
+
+A Kubernetes cluster can be created using either HPE GreenLake for Containers GUI, or through Infrastructure as Code (IaC) with [HPE GreenLake Terraform Provider](https://registry.terraform.io/providers/HPE/hpegl/0.2.2). By launching to the HPE GreenLake for Containers service console, you can download the **kubectl** binary, together with the *kubeconfig* file, and set it up to access the Kubernetes cluster using kubectl CLI. The administrative access will be configured from the created Kubernetes cluster in HPE GreenLake for containers, which allows you to set up the Kubernetes RBAC for MongoDB application deployment.
```

Comment:
You may want to also make reference to the blog post https://developer.hpe.com/blog/kubernetes-clusters-as-code-part1/ 

A Kubernetes cluster can be created.....with [HPE GreenLake Terraform](https://registry.terraform.io/providers/HPE/hpegl/0.2.2) as explained in the blog post [Kubernetes Cluster as Code](https://developer.hpe.com/blog/kubernetes-clusters-as-code-part1/).